compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs

```diff
@@ -222,8 +222,8 @@ impl ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
                                 let id_to_set = *ids.iter().min().unwrap();
 
                                 // Sort the id list so that the algorithm is deterministic
-                                let mut ids = ids.into_iter().collect::<SmallVec<[_; 8]>>();
-                                ids.sort();
+                                let mut ids = ids.into_iter().collect::<SmallVec<[usize; 8]>>();
+                                ids.sort_unstable();
```

Comment:
Perhaps this (and lower) `sort`'s need to be stable?

Reply:
We are sorting primitives (`usize`s) here, so it does not matter if we have `1, 1, 1` or `1, 1, 1` :)